payments-core/src/main/java/com/stripe/android/payments/PaymentFlowResultProcessor.kt

```diff
@@ -371,34 +374,43 @@ internal class SetupIntentFlowResultProcessor @Inject constructor(
         clientSecret: String,
         requestOptions: ApiRequest.Options,
         expandFields: List<String>
-    ): SetupIntent? =
-        stripeRepository.retrieveSetupIntent(
-            clientSecret,
-            requestOptions,
-            expandFields
+    ): Result<SetupIntent> = runCatching {
+        requireNotNull(
+            stripeRepository.retrieveSetupIntent(
+                clientSecret,
+                requestOptions,
+                expandFields
+            )
         )
+    }
 
     override suspend fun refreshStripeIntent(
         clientSecret: String,
         requestOptions: ApiRequest.Options,
         expandFields: List<String>
-    ): SetupIntent? =
-        stripeRepository.retrieveSetupIntent(
-            clientSecret,
-            requestOptions,
-            expandFields
+    ): Result<SetupIntent> = runCatching {
+        requireNotNull(
+            stripeRepository.retrieveSetupIntent(
+                clientSecret,
+                requestOptions,
+                expandFields
+            )
         )
+    }
 
     override suspend fun cancelStripeIntentSource(
         stripeIntentId: String,
         requestOptions: ApiRequest.Options,
         sourceId: String
-    ): SetupIntent? =
-        stripeRepository.cancelSetupIntentSource(
-            stripeIntentId,
-            sourceId,
-            requestOptions
+    ): Result<SetupIntent> = runCatching {
```

Comment:
This pattern (`runcatching requireNotNull`) is used a lot throughout. Should we add a helper for this?

Also, I'm not sure I like this pattern 🙈 Is this the desired end state?